### ISSUE TITLE
[exporter/file]

### ISSUE THREAD

**Issue by djaglowski** [open-telemetry/opentelemetry-collector-contrib#14690]
## Description
`rotation` settings were added in v0.60.0. Unfortunately, they are enabled by default, which was an undeclared breaking change.

I'm opening this issue to discuss whether the previous functionality should be restored.

## Steps to Reproduce
Configure the file exporter without specifying rotation settings.

## Expected Result
File rotation is not enabled.

## Actual Result
File rotation is enabled.

### Collector version

v0.60.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### Additional context

The intended design would have file rotation enabled only if the user specified at least `rotation:` in the config. However, if specified, related default settings would apply. This would allow users to enable rotation very easily without having to specify specific settings.

Since the behavior has already changed, the question should be asked - should it be corrected in the next version, to behave as described above? Alternately, should it stay ie. is there a strong case to be made that rotation should be enabled by default?

**Comment by djaglowski** [comment]
@atingchen, I'm curious your thoughts on this.

**Comment by atingchen** [comment]
@djaglowski  Sorry for causing this breaking change.
When the amount of data exported by the user is small or only some data is exported each time, I think the performance of the rotation exporting a 100MB file by default is the same as before.
When the amount of data exported from user data is large or grows infinitely, the following problems may occur:
- Larger files are harder to manipulate.
- File systems run out of space.

For these reasons, I set rotation as the default.
In order to avoid problems that may be caused to users by the infinite growth of files, should we set rotation as default and provide a flag to disable it？

**Comment by djaglowski** [comment]
I am in favor of having rotation disabled by default. However, if possible, we should find a way to do this that does not require a flag. My ideal would be that it behaves like this:

```yaml
file/no_rotation:
  path: ./foo

file/rotation_with_default_settings:
  path: ./foo
  rotation:

file/rotation_with_custom_settings:
  path: ./foo
  rotation:
    max_megabytes: 1234
    ...
```

**Comment by atingchen** [comment]
I'll open a PR to fix this.

**Comment by djaglowski** [comment]
Thanks @atingchen

**Comment by evan-bradley** [comment]
Closing as resolved via #14705.
